parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestEncryptionOptions.java

```diff
@@ -0,0 +1,660 @@
+        Path file = new Path(listOfFiles[fileNum].getAbsolutePath());
+        if (!file.getName().endsWith("parquet.encrypted")) { // Skip non encrypted files
+          continue;
+        }
+        EncryptionConfiguration encryptionConfiguration = getEncryptionConfigurationFromFilename(file.getName());
+        if (null == encryptionConfiguration) {
+          continue;
+        }
+        LOG.info("--> Read file {} {}", file.toString(), encryptionConfiguration);
+
+        // Read only the non-encrypted columns
+        if ((decryptionConfiguration == DecryptionConfiguration.NO_DECRYPTION) &&
+          (encryptionConfiguration == EncryptionConfiguration.ENCRYPT_COLUMNS_PLAINTEXT_FOOTER)) {
+          conf.set("parquet.read.schema", Types.buildMessage()
+            .required(BOOLEAN).named(BOOLEAN_FIELD_NAME)
+            .required(INT32).named(INT32_FIELD_NAME)
+            .named("FormatTestObject").toString());
+        }
+
+        ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), file)
+          .withConf(conf)
+          .withDecryption(fileDecryptionProperties)
+          .build();
+
+        try {
+          for (int i = 0; i < 500; i++) {
+            Group group = null;
+            group = reader.read();
+            boolean expect = false;
+            if ((i % 2) == 0)
+              expect = true;
+            boolean bool_res = group.getBoolean(BOOLEAN_FIELD_NAME, 0);
+            if (bool_res != expect)
+              addErrorToErrorCollectorAndLog("Wrong bool", encryptionConfiguration, decryptionConfiguration);
+            int int_res = group.getInteger(INT32_FIELD_NAME, 0);
+            if (int_res != i)
+              addErrorToErrorCollectorAndLog("Wrong int", encryptionConfiguration, decryptionConfiguration);
+            if (decryptionConfiguration != DecryptionConfiguration.NO_DECRYPTION) {
+              float float_res = group.getFloat(FLOAT_FIELD_NAME, 0);
+              float tmp1 = (float) i * 1.1f;
+              if (float_res != tmp1)
+                addErrorToErrorCollectorAndLog("Wrong float", encryptionConfiguration, decryptionConfiguration);
+
+              double double_res = group.getDouble(DOUBLE_FIELD_NAME, 0);
+              double tmp = (i * 1.1111111);
+              if (double_res != tmp)
+                addErrorToErrorCollectorAndLog("Wrong double", encryptionConfiguration, decryptionConfiguration);
+            }
+          }
+        } catch (Exception e) {
+          String errorMessage = e.getMessage();
+          checkResult(file.getName(), decryptionConfiguration, (null == errorMessage ? "" : errorMessage));
+        }
+        conf.unset("parquet.read.schema");
+      }
+    }
+  }
+
+
+  /**
+   * Create a number of Encryption configurations
+   * @param AADPrefix
+   * @return
+   */
+  private Map<EncryptionConfiguration, FileEncryptionProperties> getEncryptionConfigurations(byte[] AADPrefix) {
```

Comment:
You may shorten this method an maybe create more readable code if you convert the _switch-case_ to a method implementation inside the `EncryptionConfiguration` enum.